### ISSUE TITLE
[bug]: Uploading files results in 500 (Failed to complete multipart upload)

### ISSUE THREAD

**Issue by FoxtrotPerry** [pingdotgg/uploadthing#522]
### Provide environment information

```bash
System:
    OS: Linux 5.15 Ubuntu 20.04.5 LTS (Focal Fossa)
    CPU: (16) x64 AMD Ryzen 7 5800X 8-Core Processor
    Memory: 10.99 GB / 15.58 GB
    Container: Yes
    Shell: 3.4.1 - /usr/bin/fish
  Binaries:
    Node: 20.10.0 - ~/.nvm/versions/node/v20.10.0/bin/node
    npm: 10.2.3 - ~/.nvm/versions/node/v20.10.0/bin/npm
    pnpm: 8.11.0 - ~/.nvm/versions/node/v20.10.0/bin/pnpm
  npmPackages:
    @uploadthing/react: ^6.0.2 => 6.0.2 
    typescript: ^5.1.6 => 5.1.6 
    uploadthing: ^6.1.0 => 6.1.0
```

### Describe the bug

Hey!

I'm new to the platform so I'm crossing my fingers hoping there's just something that I'm missing but I was able to reproduce the issue with the minimal next pages demo as see in the link below.

## The error

TLDR: Trying to upload any file results in an error regardless of configuration.

When attempting to upload files in my application (or in the minimal demo as previously stated) I'm consistently getting the following error in my console:

![image](https://github.com/pingdotgg/uploadthing/assets/31854658/427ab582-154c-4d72-8905-d746c84e6518)

This error ends up leaving files in a permanent "Uploading" state:

![image](https://github.com/pingdotgg/uploadthing/assets/31854658/5eafc474-f44b-48b5-9e03-835e543246f5)

And for the sake of having everything in one place from the start, here's what the browser reports as well:

![image](https://github.com/pingdotgg/uploadthing/assets/31854658/5dd3ef06-9c78-4363-af9f-e009d2926b42)

### Link to reproduction

https://github.com/FoxtrotPerry/ut-issue-demo

### To reproduce

1. Pull down repo (or clone from example project as it's identical besides the burner env vars on my fork)
2. Boot up, and attempt to upload a file.
3. If issue is still present, you should be seeing what's described in this issue at this point.

### Additional information

I was able to get files in the same state as seen on my dashboard by uploading them through the dashboard itself so that's definitely a large contributing factor to why I'm not hopeful that this is something silly I missed on my side.

![image](https://github.com/pingdotgg/uploadthing/assets/31854658/dc596ef7-0897-4bb2-be46-27d852e697b1)

Metadata for the photo used for testing:
- png
- ~300KB
- 576 x 382

_almost definitely doesn't matter but better to have more than than is needed than too little_

### 👨‍👧‍👦 Contributing

- [X] 🙋‍♂️ Yes, I'd be down to file a PR fixing this bug!

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by juliusmarminge** [comment]
Can you add an error formatter and provide us with the logs it outputs on your server?

```ts
const f = createUploadthing({
  /**
   * Log out more information about the error, but don't return it to the client
   * @see https://docs.uploadthing.com/errors#error-formatting
   */
  errorFormatter: (err) => {
    console.log("Error uploading file", err.message);
    console.log("  - Above error caused by:", err.cause);
    
    return { message: err.message };
  },
});
```

**Comment by FoxtrotPerry** [comment]
> Can you add an error formatter and provide us with the logs it outputs on your server?
> 
> ```ts
> const f = createUploadthing({
>   /**
>    * Log out more information about the error, but don't return it to the client
>    * @see https://docs.uploadthing.com/errors#error-formatting
>    */
>   errorFormatter: (err) => {
>     console.log("Error uploading file", err.message);
>     console.log("  - Above error caused by:", err.cause);
>     
>     return { message: err.message };
>   },
> });
> ```

Sure thing! Here's the error but with the `errorFormatter` slightly tweaked to conform to the format you posted.

![image](https://github.com/pingdotgg/uploadthing/assets/31854658/13689c28-090b-46bd-879c-3694adc8af20)

**Comment by markflorkowski** [comment]
@FoxtrotPerry  Are you able to share the particular file you were encountering this issue with? I have not been able to reproduce this error so far on my end.

**Comment by FoxtrotPerry** [comment]
> @FoxtrotPerry Are you able to share the particular file you were encountering this issue with? I have not been able to reproduce this error so far on my end.

@markflorkowski 

Absolutely! It's just a very compressed version of a photo I took for a college project back in the day (critiques on the photo welcomed as well lol)

![CEP_8688](https://github.com/pingdotgg/uploadthing/assets/31854658/7ed53841-112d-4756-9063-da017161521e)

Not sure how github's compression will modify it further so I'll give a discrete download link as well:

https://mega.nz/file/tQZx3ZCa#pz6tRTqpyeMyTWAgCQO8d5qQrgn64sT3bj2Kf-ug6nE

**Comment by markflorkowski** [comment]
Hmm yeah, still not able to repro this. Uploads fine for me both on dash and with a local project. 

![image](https://github.com/pingdotgg/uploadthing/assets/11494384/d83f94d0-48c9-4a6d-ab6c-917c2e62a49d)

**Comment by juliusmarminge** [comment]
Same: https://utfs.io/f/9cf6f8f8-c488-4ce9-9f23-62d9de2afd1c-1ntw1.png

**Comment by FoxtrotPerry** [comment]
Thank you both for looking into this I really appreciate it! I tried it again but this time with all four pertinent pieces of info up and visible. In the video I attempt to upload via the minimal example and once more (with the same file) from the dashboard.

https://github.com/pingdotgg/uploadthing/assets/31854658/b45d6cca-c264-4dc2-a004-4bc02f5195cd

Give that you two are able to upload with the example just fine I'm now thinking there's a possibility that it could be a browser issue? Not sure. I'm using Brave with a few different addons so I'm going to try first an incognito window to see if I can reproduce the issue and then if that still fails, I'll re-attempt with a fresh install of some chromium based browser.

Let me know if there's anything else I can provide in helping track this issue down! And if for whatever reason the video is too compressed to see anything clearly that might be important let me know and I can upload it to mega as well.

**Comment by markflorkowski** [comment]
It definitely could be browser related, especially since I know Brave is supposed to be one of the ones with built in adblocking and such, which could mess with network traffic in unpredictable ways. I was testing on Arc on a Mac. Will try to find a PC to test on to see if there is any difference, but I think it is more likely due to Brave than Windows.

**Comment by FoxtrotPerry** [comment]
@markflorkowski @juliusmarminge 

I have an update!

I've tested three different browsers across macos, linux, and windows and found something interesting. TLDR: It's related to the extension [ClearURLs](https://docs.clearurls.xyz/1.26.1/)

What you'll see in the video is me:
- Leave ClearURLs off, upload a file
- Wait for it to upload sucessfully
- Turn ClearURLs on, upload another file
- Wait the same amount of time giving it plenty of time to succeed
- Turn ClearURLs back off and upload another file
- Notice that the third file beats the second to upload

https://github.com/pingdotgg/uploadthing/assets/31854658/4b6ae094-dcfe-4676-9797-ed249abd5998

I'm going to leave that extension off and go test the minimal example. If that works as anticipate then I think we can say we cracked the case and close this out. Although I'll say that I have never had this problem with this extension turned on with any other upload service so maybe root cause analysis would still be something worth delving into.

Either way, I'll report back!

**Comment by FoxtrotPerry** [comment]
Just tried the minimal example, everything is working as expected!

Just for kicks I tried permutations with privacy badger and ublock origin turned on as well and as you'd expect uploading works with them enabled as well. It seems to be just ClearURLs that is the trouble maker.

**Comment by markflorkowski** [comment]
Yup, looks like others have encountered this before as well: https://github.com/ClearURLs/Addon/issues/177
